aws_marketplace/using-open-source-model-packages/pytorch-ic-model/notebook.ipynb

```diff
@@ -0,0 +1,461 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perform_inference(cat_jpg)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perform_inference(dog_jpg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### D. Delete the endpoint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that you have successfully performed a real-time inference, you do not need the endpoint any more. You can terminate the endpoint to avoid being charged."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.sagemaker_session.delete_endpoint(model_name)\n",
+    "model.sagemaker_session.delete_endpoint_config(model_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Perform batch inference"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this section, you will perform batch inference using multiple input payloads together. If you are not familiar with batch transform, and want to learn more, see these links:\n",
+    "1. [How it works](https://docs.aws.amazon.com/sagemaker/latest/dg/ex1-batch-transform.html)\n",
+    "2. [How to run a batch transform job](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-batch.html)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#upload the batch-transform job input files to S3\n",
+    "transform_input_key_prefix = 'image-classification-model-transform-input'\n",
+    "transform_input = sagemaker_session.upload_data(cat_jpg, key_prefix=transform_input_key_prefix) \n",
+    "print(\"Transform input uploaded to \" + transform_input)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Run the batch-transform job\n",
+    "transformer = model.transformer(1, batch_transform_inference_instance_type)\n",
+    "transformer.transform(transform_input, content_type=content_type)\n",
+    "transformer.wait()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# output is available on following path\n",
+    "transformer.output_path"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_location = '{}/{}'.format(transformer.output_path, 'cat.jpg.out')\n",
+    "batch_prediction = \"\"\n",
+    "\n",
+    "# print out batch-transform job output\n",
+    "for line in smart_open(data_location, 'rb'):\n",
```

Comment:
You could reduce compatibility and portability issues by removing this dependency.